pallets/public-credentials/src/lib.rs

```diff
@@ -585,6 +594,31 @@ pub mod pallet {
 				}
 			})
 		}
+
+		#[cfg(any(feature = "try-runtime", test))]
+		pub fn do_try_state() -> DispatchResult {
+			// check if for each owner there is a name stored.
+			Credentials::<T>::iter().try_for_each(|(subject_id, credential_id, entry)| -> DispatchResult {
+				ensure!(
+					CredentialSubjects::<T>::contains_key(&credential_id),
+					DispatchError::Other("Test")
+				);
+
+				ensure!(
+					CredentialSubjects::<T>::get(&credential_id).unwrap() == subject_id,
```

Comment:
```suggestion
					CredentialSubjects::<T>::get(&credential_id) == Some(subject_id),
```

Reply:
This is enough, and allows you to remove the check above. This should be applied to all the other places as well.